GeeksCoreLibrary/Core/Models/PermissionModel.cs

```diff
@@ -27,4 +27,14 @@ public class PermissionModel
     /// Gets or sets the actual permissions. This is an enum with the <see cref="FlagsAttribute"/>, which means you can set multiple permissions in one property.
     /// </summary>
     public AccessRights Permissions { get; set; }
+
+    /// <summary>
+    /// Gets or sets the URL for the endpoint.
+    /// </summary>
+    public string EndpointUrl { get; set; }
+
+    /// <summary>
+    /// Gets or sets the HTTP method for the endpoint.
+    /// </summary>
+    public string EndpointHttpMethod { get; set; }
```

Comment:
As discussed please change the property type to the HTTP method enum: https://learn.microsoft.com/en-us/dotnet/api/microsoft.bot.builder.dialogs.adaptive.actions.httprequest.httpmethod?view=botbuilder-dotnet-stable

Reply:
The link you send is from some weird interface that we don't have (microsoft.bot.builder.etc). The best alternative I found is https://learn.microsoft.com/en-us/dotnet/api/system.net.http.httpmethod?view=net-8.0&redirectedfrom=MSDN#Anchor_6, which is not an enum but a class with static properties. I used that one now.